Fix SocketCategoryChannel properties

SubZero0 opened the pull request:
Probably forgotten when copied from another file.

SubZero0 committed:
Update Users property for category channels

FiniteReality commented on src/Discord.Net.WebSocket/Entities/Channels/SocketCategoryChannel.cs:
```diff
@@ -15,7 +15,7 @@ namespace Discord.WebSocket
     {
         public override IReadOnlyCollection<SocketGuildUser> Users
-            => Guild.Users.Where(x => x.VoiceChannel?.Id == Id).ToImmutableArray();
+            => throw new NotSupportedException();
```
Wouldn't it be better to use something like SocketTextChannels.Users, and resolve the permissions instead of throwing an exception?

SubZero0 replied:
Hm, makes sense. I'll update it.

SubZero0 replied:
As talked in the dev guild, resolving the permissions could be misleading, since if the users don't have read access to any channels inside, they won't be able to see the category too.

A suggestion would be `Guild.Channels.Where(x => x.CategoryId == Id).SelectMany(x => x.Users).Distinct().ToImmutableArray();` that would list all users that can see at least one channel (and the category consequently) but doesn't mean all of them can see all the channels inside it.

SubZero0 committed:
Wrong property being used for Channels property

CategoryId is the category that "owns" this channel. That is actually impossible right now for category channels, so it returns null and get all channels wrongly.

SubZero0 committed:
Resolve permissions for category

SubZero0 committed:
Remove spaces

SubZero0 commented:
Just a reminder that now this PR depends on #952

SubZero0 committed:
Small fix for IChannel.GetUsersAsync
